python/src/iceberg/utils/decimal.py

```diff
@@ -75,3 +75,16 @@ def decimal_to_bytes(value: Decimal) -> bytes:
     """
     unscaled_value = decimal_to_unscaled(value)
     return unscaled_value.to_bytes(bytes_required(unscaled_value), byteorder="big", signed=True)
+
+
+def truncate_decimal(value: Decimal, width: int) -> Decimal:
+    """Get a truncated Decimal value given a decimal value and a width
+    Args:
+        value (Decimal): a decimal value
+        width (int): A width for the returned Decimal instance
+    Returns:
+        Decimal: A truncated Decimal instance
+    """
+    unscaled_value = decimal_to_unscaled(value)
+    applied_value = unscaled_value - (((unscaled_value % width) + width) % width)
```

Comment:
This expression doesn't need to be as complex. The purpose of this is to handle negative numbers in languages where `%` will return a negative value. Python returns positive values:

```python
-4 % 5                   # => 1
((-4 % 5) + 5) % 5 # => 1
```

Note that if `-4 % 5` resulted in `-4` because |-4| < 5 then we would need the more complex expression.